extensions/scarb-snforge-test-collector/src/metadata.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, Context, Result};
+use cairo_lang_filesystem::db::Edition;
+use cairo_lang_project::{AllCratesConfig, SingleCrateConfig};
+use cairo_lang_utils::ordered_hash_map::OrderedHashMap;
+use camino::Utf8PathBuf;
+use scarb_metadata::{CompilationUnitMetadata, Metadata, PackageMetadata};
+use smol_str::SmolStr;
+use std::path::PathBuf;
+
+/// Represents a dependency of a Cairo project
+#[derive(Debug, Clone)]
+pub struct LinkedLibrary {
+    pub name: String,
+    pub path: PathBuf,
+}
+
+pub fn corelib_for_compilation_unit(
+    compilation_unit: &CompilationUnitMetadata,
+) -> Result<Utf8PathBuf> {
+    let corelib = compilation_unit
+        .components
+        .iter()
+        .find(|du| du.name == "core")
+        .context("corelib could not be found")?;
+    Ok(Utf8PathBuf::from(corelib.source_root()))
+}
+
+pub fn dependencies_for_compilation_unit(
+    compilation_unit: &CompilationUnitMetadata,
+) -> Vec<LinkedLibrary> {
+    let dependencies = compilation_unit
+        .components
+        .iter()
+        .filter(|du| &du.name != "core")
+        .map(|cu| LinkedLibrary {
+            name: cu.name.clone(),
+            path: cu.source_root().to_owned().into_std_path_buf(),
+        })
+        .collect();
+
+    dependencies
+}
+
+pub fn crates_config_for_compilation_unit(
+    scarb_metadata: &Metadata,
+    compilation_unit: &CompilationUnitMetadata,
+) -> AllCratesConfig {
+    let crates_config: OrderedHashMap<SmolStr, SingleCrateConfig> = compilation_unit
+        .components
+        .iter()
+        .map(|component| {
+            (
+                SmolStr::from(&component.name),
+                SingleCrateConfig {
+                    edition: if let Some(edition) = scarb_metadata
+                        .get_package(&component.package)
+                        .unwrap()
+                        .edition
+                        .clone()
+                    {
+                        serde_json::from_str(&edition).unwrap()
+                    } else {
+                        Edition::default()
+                    },
+                },
+            )
+        })
+        .collect();
+
+    AllCratesConfig {
+        override_map: crates_config,
+        ..Default::default()
+    }
+}
+
+pub fn compilation_unit_for_package<'a>(
+    metadata: &'a Metadata,
+    package_metadata: &PackageMetadata,
+) -> Result<&'a CompilationUnitMetadata> {
+    metadata
+        .compilation_units
+        .iter()
+        .filter(|unit| unit.package == package_metadata.id)
+        .min_by_key(|unit| match unit.target.kind.as_str() {
```

Comment:
Just to make sure - we should use `kind` here? `name` was used before 🤔

Reply:
Yes, makes sense to use kind.